subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/BaselineVersion.groovy

```diff
@@ -56,19 +59,39 @@ class BaselineVersion implements VersionResults {
             } else {
                 sb.append "Speed $displayName: Results were inconclusive"
             }
-            String confidencePercent = confidenceInDifference(results.totalTime, current.totalTime) * 100 as int
-            sb.append(" with " + confidencePercent + "% confidence.\n")
+            Number regressionPercentage = PrettyCalculator.percentChange(currentVersionMean, thisVersionMean)
+            String confidence = "${confidenceInDifference(results.totalTime, current.totalTime) * 100 as int}%"
+            sb.append(" with $confidence confidence.\n")
 
             def diff = currentVersionMean - thisVersionMean
             def desc = diff > Duration.millis(0) ? "slower" : "faster"
-            sb.append("Difference: ${diff.abs().format()} $desc (${toMillis(diff.abs())}), ${PrettyCalculator.percentChange(currentVersionMean, thisVersionMean)}%\n")
+            sb.append("Difference: ${diff.abs().format()} $desc (${toMillis(diff.abs())}), ${regressionPercentage}%\n")
             sb.append(current.speedStats)
             sb.append(results.speedStats)
             sb.append("\n")
-            sb.toString()
+            sb.append(dataForMachine(current, results, regressionPercentage.doubleValue(), confidence))
         } else {
             sb.append("Speed measurement is not available (probably due to a build failure)")
         }
+        sb.toString()
+    }
+
+    String dataForMachine(MeasuredOperationList experimentGroup, MeasuredOperationList controlGroup, double regressionPercentage, String confidence) {
+        StringBuilder sb = new StringBuilder("$MACHINE_DATA_SEPARATOR\n")
+        sb.append(new ObjectMapper().writeValueAsString(new ScenarioBuildResultData.ExperimentData(
```

Comment:
I don't think we need to write out that data - it can be re-constructed from the performance database, right?